tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/EtcExpr.scala

```diff
@@ -0,0 +1,117 @@
+package at.forsyte.apalache.tla.typecheck.etc
+
+import at.forsyte.apalache.tla.typecheck.TlaType1
+
+/**
+  * An expression in a simple typed lambda calculus. Here we do not care about the concrete values,
```

Comment:
Since TLA expressions are referred to as `TlaEx`, wouldn't it be more consistent to call these `EtcEx`? Otherwise we have two different abbreviations for the same word.

Reply:
That is a good point :-)

Reply:
I created a separate issue, so it does not get lost in this huge PR.

Reply:
Correcting the naming here won't impact the readability or review-ability of the huge PR, because you're just introducing the code for the first time.